operator/controllers/controller/reconciler.go

```diff
@@ -248,6 +268,55 @@ func (r *ControllerReconciler) updateStatus(ctx context.Context, instance *contr
 	return nil
 }
 
+func (r *ControllerReconciler) deleteSingletonResources(ctx context.Context, log logr.Logger, req ctrl.Request, instance *controllerv1alpha1.Controller) error {
+	// If we created controller with name "controller" its full name will match singleton name.
+	instanceFullName := fmt.Sprintf("%s-%s", controllers.AppName, instance.GetName())
+	if instanceFullName == controllers.ControllerServiceName {
```

Comment:
This should be `!=`, right? As we are trying to do deletion for resources with older names after this change? Or I misunderstood this function?

Reply:
We want to avoid deletion of the resource in case instanceFullName matches controllers.ControllerServiceName, otherwise reconcile loop will constantly delete and recreate it.